app/src/tracing.h

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "constants.h"
+
+#include <time.h>
+#include <regex>
+#include <sstream>
+#include <functional>
+#include <ccf/endpoint_context.h>
+#include <ccf/base_endpoint_registry.h>
+
+namespace scitt
+{
+  std::string REQUEST_ID_HEADER = "x-ms-request-id";
+  std::string CLIENT_REQUEST_ID_HEADER = "x-ms-client-request-id";
```

Comment:
Azure SDK guidelines already advocate for `traceparent` and `tracestate` .
* https://azure.github.io/azure-sdk/general_azurecore.html#distributed-tracing-policy

API guidelines suggest followng the SDK guidelines but also mention common headers (implemented ATM) but it is mentioned like:

> In addition to distributed tracing, Azure also uses a set of common correlation headers

* https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#distributed-tracing--telemetry

Reply:
In another place [in the guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#http-request--response-pattern):

> The table below lists the headers most used by Azure services:
> ...
> _x-ms-request-id_ | Response 
> ...
> DO support all headers shown in _italics_

I suppose `x-ms-request-id` needs to be returned from the ledger ATM. But

Reply:
We can implement W3C trace context for the next release if needed but it's a bit more work than the two x-ms headers, therefore I didn't do it here. The main focus is on improving the logging for this PR and have some correlation identifiers.